lms/views/decorators/h_api.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+
+"""View decorators for working with the h API."""
+
+from __future__ import unicode_literals
+
+import json
+import requests
+
+from pyramid.httpexceptions import HTTPBadGateway
+from pyramid.httpexceptions import HTTPBadRequest
+from pyramid.httpexceptions import HTTPGatewayTimeout
+
+from lms import util
+from lms.util import MissingToolConsumerIntanceGUIDError
+from lms.util import MissingUserIDError
+
+
+def create_h_user(wrapped):  # noqa: MC0001
+    """
+    Create a user in h if one doesn't exist already.
+
+    Call the h API to create a user for the authorized LTI user, if one doesn't
+    exist already.
+    """
+    def wrapper(request, jwt):  # pylint: disable=too-many-branches
+        params = request.params
+
+        try:
+            oauth_consumer_key = params["oauth_consumer_key"]
+        except KeyError:
+            raise HTTPBadRequest('Required paramer "oauth_consumer_key" missing from LTI params')
+
+        # Only create users for application instances that we've enabled the
+        # auto provisioning features for.
+        enabled_consumer_keys = request.registry.settings["auto_provisioning"]
+        if oauth_consumer_key not in enabled_consumer_keys:
+            return wrapped(request, jwt)
+
+        # Our OAuth 2.0 client_id and client_secret for authenticating to the h API.
+        client_id = request.registry.settings["h_client_id"]
+        client_secret = request.registry.settings["h_client_secret"]
+
+        # The authority that we'll create h users and groups in.
+        authority = request.registry.settings["h_authority"]
+
+        # The URL of h's create user API endpoint.
+        create_user_api_url = request.registry.settings["h_api_url"] + "/users"
```

Comment:
All these config settings are required config settings, so we can safely assume they're present